test/overseer/store_test.clj

```diff
@@ -24,6 +24,20 @@
   (->> (tcore/minus now (tcore/millis millis-ago))
        std.time/datetime->unix))
 
+(defn test-transact-graph [store]
+  (testing "idempotent transaction"
+    (let [j1 (test-utils/job {:job/id "foo-id" :job/args {:x 1}})
+          graph1 (api/simple-graph j1)
+
+          j2 (test-utils/job {:job/id "foo-id" :job/args {:x 2}})
+          graph2 (api/simple-graph j2)]
+      (is (= graph1 (core/transact-graph store graph1))
+          "It returns the graph after inserting into the store")
+      (is (= graph2 (core/transact-graph store graph2))
+          "It is idempotent on duplicate job-id insert")
+      (is (= {:x 1} (:job/args (core/job-info store "foo-id")))
+          "It does not double-insert into the store"))))
```

Comment:
I think this would be clearer if it side effected transacting `graph2` then tested for the arg, rather than this expectation being split across two `is` clauses.